(test)O3-2111: Add patient chart add program e2e test

RandilaP opened the pull request:
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
Goal of this PR is to add a e2e test of add program function.
<!-- Please describe what problems your PR addresses. -->

## Screenshots
<img width="1680" alt="image" src="https://github.com/openmrs/openmrs-esm-patient-chart/assets/106733522/f2a2bc2c-b379-447f-b2f4-ba74e4abe15f">

<!-- Required if you are making UI changes. -->

## Related Issue
<!-- Paste the link to the Jira ticket here if one exists. -->
https://issues.openmrs.org/browse/O3-2111

## Other
<!-- Anything not covered above -->

RandilaP commented:
Hey guys. Could you please take a look at this? I need some help with adding a data test id and I like to ask your opinion about what I'm missing @anjula-sack @Piumal1999 @jayasanka-sack

anjula-sack commented on e2e/specs/add-program.spec.ts:
```diff
@@ -0,0 +1,40 @@
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+});
```
```suggestion
});

```

anjula-sack commented on e2e/specs/add-program.spec.ts:
```diff
@@ -0,0 +1,40 @@
+  await test.step('Then I should see the program in the patient chart', async () => {
+    await expect(programPage.tableRow().getByText('PMTCT')).toBeVisible();
+  });
+});
```
```suggestion
});

```

anjula-sack commented on packages/esm-patient-programs-app/src/programs/programs-overview.component.tsx:
```diff
@@ -159,7 +159,7 @@ const ProgramsOverview: React.FC<ProgramsOverviewProps> = ({ basePath, patientUu
                   </TableRow>
                 </TableHead>
-                <TableBody>
+                <TableBody data-testid="playwright1">
```
Let's remove the unwanted test ids @RandilaP

RandilaP commented:
@jayasanka-sack @anjula-sack @Piumal1999 I updated this PR with the current requirements that you asked for in previous PRs. I couldn't test this in the local server because of the issues with the form. But I tested this using the dev server. After testing it I added test ids and used them. I also locate each cells like in the allergies PR. I updated this test in to add a program and edit a program as @anjula requested

anjula-sack commented on packages/esm-patient-programs-app/src/programs/programs-detailed-summary.component.tsx:
```diff
@@ -139,14 +139,14 @@ const ProgramsDetailedSummary: React.FC<ProgramsDetailedSummaryProps> = ({ patie
                       ))}
-                      <TableCell className="cds--table-column-menu">
-                        <ProgramEditButton programEnrollmentId={enrollments[i]?.uuid} />
+                      <TableCell className="cds--table-column-menu" data-testid={'button'}>
```
Is this required? @RandilaP

RandilaP commented:
@jayasanka-sack Could you please review this

anjula-sack commented on e2e/specs/add-and-edit-program.spec.ts:
```diff
@@ -0,0 +1,80 @@
+  const programPage = new ProgramPage(page);
+  const row = programPage.table().locator('tr');
+  const programCell = row.locator('td:first-child');
+  const LocationCell = row.locator('td:nth-child(2)');
```
```suggestion
  const locationCell = row.locator('td:nth-child(2)');
```

anjula-sack reviewed (approved):
Good job! @RandilaP
